qiskit/ignis/verification/randomized_benchmarking/fitters.py

```diff
@@ -750,3 +763,189 @@ def plot_rb_data(self, pattern_index=0, ax=None,
 
         if show_plt:
             plt.show()
+
+
+class PurityRBFitter(RBFitterBase):
+    """
+        Class for fitters for purity RB
+        Derived from RBFitterBase class
+
+        Contains 3^n+1 RBFitter objects:
+        3^n RBFitter objects:
+        to calculate the raw_data from the results
+        and another RBFitter object:
+        calculates their sum and does the fit
+    """
+
+    def __init__(self, purity_result, npurity, cliff_lengths,
+                 rb_pattern=None):
+        """
+        Args:
+            purity_result: list of lists of results of the
+            3^n purity RB sequences per seed (qiskit.Result).
+            npurity: equals 3^n (where n is the dimension)
+            cliff_lengths: the Clifford lengths, 2D list i x j where i is the
+                number of patterns, j is the number of cliffords lengths
+            rb_pattern: the pattern for the rb sequences.
+        """
+
+        self._cliff_lengths = cliff_lengths
+        self._rb_pattern = rb_pattern
+        self._npurity = npurity
+        self._fit = [{} for e in rb_pattern]
+
+        self._rbfit_purity = [[] for _ in range(npurity+1)]
+        # 3^n RBFitter objects
+        for d in range(self._npurity):
```

Comment:
I don't think it's going to work to try and use 3^n RBFitter objects (see comments below)

Reply:
The aim of using 3^n fitters is in order to re-use the code of add_data and calc_data. 
Basically, after calculation `raw_data`, I don't need them anymore.

Reply:
OK - I changed this